js/pin.js

```diff
@@ -16,21 +16,67 @@
     return pinElement;
   }
 
-  var resultXHR = {};
-  function successHandler(author) {
-    author.forEach(function (el, index) {
-      fragment.appendChild(makeElement(el, index));
-      resultXHR[index] = author [index];
+  var sameresultXHR = [];
+
+  function filter(arr) {
+
+    var randomResult = shuffle(arr);
+    sameresultXHR = randomResult.filter(function (it, index) {
+      return index < 5;
     });
```

Comment:
```suggestion
    sameresultXHR = randomResult.filter(function (it, index) {
      return index < 5;
    });
```

А если тут будет 100500 элементов, то фильтр пройдёт по всем

ЛУчше посмотри в сторону оператора `slice`